src/OmniSharp.Roslyn.CSharp/Services/Refactoring/V2/BaseCodeActionService.cs

```diff
@@ -23,41 +25,63 @@ public abstract class BaseCodeActionService<TRequest, TResponse> : RequestHandle
 
         private readonly CodeActionHelper _helper;
 
+        private readonly MethodInfo _getNestedCodeActions;
+
         protected BaseCodeActionService(OmniSharpWorkspace workspace, CodeActionHelper helper, IEnumerable<ICodeActionProvider> providers, ILogger logger)
         {
             this.Workspace = workspace;
             this.Providers = providers;
             this.Logger = logger;
             this._helper = helper;
+
+            // Sadly, the CodeAction.NestedCodeActions property is still internal.
+            var nestedCodeActionsProperty = typeof(CodeAction).GetProperty("NestedCodeActions", BindingFlags.NonPublic | BindingFlags.Instance);
+            if (nestedCodeActionsProperty == null)
+            {
+                throw new InvalidOperationException("Could not find CodeAction.NestedCodeActions property.");
+            }
+
+            this._getNestedCodeActions = nestedCodeActionsProperty.GetGetMethod(nonPublic: true);
+            if (this._getNestedCodeActions == null)
+            {
+                throw new InvalidOperationException("Could not retrieve 'get' method for CodeAction.NestedCodeActions property.");
+            }
         }
 
         public abstract Task<TResponse> Handle(TRequest request);
 
-
-        protected async Task<IEnumerable<CodeAction>> GetActionsAsync(ICodeActionRequest request)
+        protected async Task<IEnumerable<AvailableCodeAction>> GetAvailableCodeActions(ICodeActionRequest request)
         {
-            var actions = new List<CodeAction>();
             var originalDocument = this.Workspace.GetDocument(request.FileName);
             if (originalDocument == null)
             {
-                return actions;
+                return Array.Empty<AvailableCodeAction>();
             }
 
-            var refactoringContext = await GetRefactoringContext(originalDocument, request, actions);
-            if (refactoringContext != null)
-            {
-                await CollectRefactoringActions(refactoringContext.Value);
-            }
+            var actions = new List<CodeAction>();
 
             var codeFixContext = await GetCodeFixContext(originalDocument, request, actions);
             if (codeFixContext != null)
             {
                 await CollectCodeFixActions(codeFixContext.Value);
             }
 
+            var refactoringContext = await GetRefactoringContext(originalDocument, request, actions);
+            if (refactoringContext != null)
+            {
+                await CollectRefactoringActions(refactoringContext.Value);
+            }
+
+            // TODO: Determine good way to order code actions.
             actions.Reverse();
```

Comment:
We have some notion of ordering in Roslyn using attributes but [it's messy](https://github.com/dotnet/roslyn/blob/master/src/Features/CSharp/Portable/CodeFixes/GenerateEnumMember/GenerateEnumMemberCodeFixProvider.cs#L18). At some point, I intend to order them similarly in OmniSharp, but it's down the priority list a bit. For now, I just wanted to add a TODO to indicate that it wasn't ideal. :smile: